src/wp-includes/blocks.php

```diff
@@ -661,31 +661,6 @@ function _excerpt_render_inner_columns_blocks( $columns, $allowed_blocks ) {
 function render_block( $parsed_block ) {
 	global $post, $wp_query;
 
-	/**
-	 * Allows render_block() to be short-circuited, by returning a non-null value.
-	 *
-	 * @since 5.1.0
-	 *
-	 * @param string|null $pre_render   The pre-rendered content. Default null.
-	 * @param array       $parsed_block The block being rendered.
-	 */
-	$pre_render = apply_filters( 'pre_render_block', null, $parsed_block );
```

Comment:
I would say this is the only place where it is a controversial change:
- `$context` processing will always be executed even when `$pre_render` returns not `null` value.
- filters will be executed in a different order as well.

Reply:
Yeah, I umm'd and ahh'd about this one.

- Leaving it in `render_block()` allows you to short-circuit the entire code path including context and makes sense as one would expect a filter named `pre_render_block` to be called right as the `render_block` function is called.

- Moving it to `WP_Block::render()` allows you to short-circuit the rendering of nested blocks and also makes sense as one would expect a filter named `pre_render_block` to be called right before the `render_block` filter is called.

Reply:
I think what I might do is call the filter in _both_ places. That way you can short-circuit the entire code path **or** just the rendering of a nested block.

Reply:
Done in 158a095.